### DISCUSSION TITLE
Demo for how to use the carla 0.9.13 function  "User Defined Paths in Traffic Manager"

### DISCUSSION THREAD

**Discussion by kivi377** [carla-simulator/carla#4960] [Q&A]
Dear Carla developer and Carla users,

I have met a problem during using the new function " Set Route" of carla latest version 0.9.13. 

I want to just use the fllowing code to test the basic funtion.:
route = ["Right", "Straight", "Right", "Right", "Left", "Right"]
traffic_manager.set_route(vehicle_bp, route)

But in python, it shows that:
Traceback (most recent call last):
  File "C:/Sumo_model/Carla_Python/main.py", line 45, in <module>
    traffic_manager.set_route(vehicle_bp, route)
Boost.Python.ArgumentError: Python argument types in
    TrafficManager.set_route(TrafficManager, ActorBlueprint, list)
did not match C++ signature:
    set_route(class carla::traffic_manager::TrafficManager {lvalue}, class boost::shared_ptr<class carla::client::Actor>, class boost::python::list, bool empty_buffer=True)

I want to ask, why this problem appears, and could you provide some demo of this "set route" new function.

BR,
Kivi

**Comment by j-scharrenbach** [comment]
Hi,

not sure if this is still relevant for you. Maybe someone else might come across this at some time.

Instead of passing the actor's blueprint, you need to pass the actual [carla.Actor](https://carla.readthedocs.io/en/latest/python_api/#carlaactor) to the `set_route` function.

```
# somehow get the actor, e.g. as returned by world.spawn_actor(...)
route = ["Right", "Straight", "Right", "Right", "Left", "Right"]
traffic_manager.set_route(actor, route)
```

Hope this still helps :)
Jannik

**Comment by yalezhang2021** [comment]
I have the same question, and i used this func with variables actor and route
and should set it as autopilot, then, it works fine:

vehicle.set_autopilot(True)
route = [ "Straight", "Left", "Left", "Left", "Right"]
traffic_manager.set_route(vehicle, route)
